### PR TITLE
simplify bootstrap build

### PR DESCRIPTION
**Pull request by tomerd**
motivation: remove depednecy on workspace that carries over dependencies on source control and registry

changes:
* move minimal-built-tool code into swift-bootstrap
* add dependency resolution code swift-bootstrap so it does not require dependency on workspace
* cleanup bootstrap arguments for those that can be hard coded
* adjust target dependencies and cmake setup accordingly

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by neonichu** [comment]
🎉

**Comment by neonichu** [comment]
Toolchain build passed ✅

File: Sources/SPMBuildCore/BuildParameters.swift [hunk 1]
```diff
@@ -229,8 +229,8 @@ public struct BuildParameters: Encodable {
         hostTriple: Triple? = nil,
         destinationTriple: Triple? = nil,
         archs: [String] = [],
-        flags: BuildFlags,
-        xcbuildFlags: [String] = [],
+        flags: BuildFlags = .init(),
```
Comment (neonichu) [inline Sources/SPMBuildCore/BuildParameters.swift:232]: Was this intentional? So far it seems like this was explicit.

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
hmm 

```
error: terminated(127): /tmp/Miscellaneous_TestDiscovery_Simple.kop1p8/Miscellaneous_TestDiscovery_Simple/.build/x86_64-unknown-linux-gnu/debug/SimplePackageTests.xctest --dump-tests-json output:
    /tmp/Miscellaneous_TestDiscovery_Simple.kop1p8/Miscellaneous_TestDiscovery_Simple/.build/x86_64-unknown-linux-gnu/debug/SimplePackageTests.xctest: error while loading shared libraries: libswift_RegexParser.so: cannot open shared object file: No such file or directory
```

does not seem to be related?

**Comment by tomerd** [comment]
@swift-ci please smoke test linux

**Comment by tomerd** [comment]
@swift-ci please smoke test linux

**Pull request merged**